--- libraw/libraw_datastream.h.orig
+++ libraw/libraw_datastream.h
@@ -104,14 +104,14 @@
 };
 
 #ifdef WIN32
-template class DllDef std::auto_ptr<std::streambuf>;
+template class DllDef std::unique_ptr<std::streambuf>;
 #endif
 
 class DllDef  LibRaw_file_datastream: public LibRaw_abstract_datastream
 {
   protected:
-    std::auto_ptr<std::streambuf> f; /* will close() automatically through dtor */
-    std::auto_ptr<std::streambuf> saved_f; /* when *f is a subfile, *saved_f is the master file */
+    std::unique_ptr<std::streambuf> f; /* will close() automatically through dtor */
+    std::unique_ptr<std::streambuf> saved_f; /* when *f is a subfile, *saved_f is the master file */
     std::string filename;
     INT64 _fsize;
 #ifdef WIN32
--- src/libraw_datastream.cpp.orig
+++ src/libraw_datastream.cpp
@@ -79,10 +79,10 @@
         _fsize = st.st_size;
 #endif
       
-      std::auto_ptr<std::filebuf> buf(new std::filebuf());
+      std::unique_ptr<std::filebuf> buf(new std::filebuf());
       buf->open(filename.c_str(), std::ios_base::in | std::ios_base::binary);
       if (buf->is_open()) {
-        f = buf;
+        f = std::move(buf);
       }
     }
 }
@@ -94,10 +94,10 @@
       struct _stati64 st;
       if(!_wstati64(wfilename.c_str(),&st))
         _fsize = st.st_size;
-      std::auto_ptr<std::filebuf> buf(new std::filebuf());
+      std::unique_ptr<std::filebuf> buf(new std::filebuf());
       buf->open(wfilename.c_str(), std::ios_base::in | std::ios_base::binary);
       if (buf->is_open()) {
-        f = buf;
+        f = std::move(buf);
       }
     }
 }
@@ -196,15 +196,15 @@
 {
     LR_STREAM_CHK();
     if (saved_f.get()) return EBUSY;
-    saved_f = f;
-        std::auto_ptr<std::filebuf> buf(new std::filebuf());
+    saved_f = std::move(f);
+        std::unique_ptr<std::filebuf> buf(new std::filebuf());
         
         buf->open(fn, std::ios_base::in | std::ios_base::binary);
         if (!buf->is_open()) {
-            f = saved_f;
+            f = std::move(saved_f);
             return ENOENT;
         } else {
-            f = buf;
+            f = std::move(buf);
         }
         
         return 0;
@@ -215,15 +215,15 @@
 {
 	LR_STREAM_CHK();
 	if (saved_f.get()) return EBUSY;
-	saved_f = f;
-	std::auto_ptr<std::filebuf> buf(new std::filebuf());
+	saved_f = std::move(f);
+	std::unique_ptr<std::filebuf> buf(new std::filebuf());
 
 	buf->open(fn, std::ios_base::in | std::ios_base::binary);
 	if (!buf->is_open()) {
-		f = saved_f;
+		f = std::move(saved_f);
 		return ENOENT;
 	} else {
-		f = buf;
+		f = std::move(buf);
 	}
 
 	return 0;
@@ -234,7 +234,7 @@
 void LibRaw_file_datastream::subfile_close()
 { 
     if (!saved_f.get()) return; 
-    f = saved_f;   
+    f = std::move(saved_f);   
 }
 
 #undef LR_STREAM_CHK
